File: backend/app/models/quest.py
from pydantic import BaseModel
from typing import List, Optional, Union
from enum import Enum

class QuestionType(str, Enum):
    QUANTITATIVE = "QUANTITATIVE"
    SOUL_NARRATIVE = "SOUL_NARRATIVE"

class VisualFeedback(str, Enum):
    GLOW_EFFECT = "GLOW_EFFECT"
    PULSE_EFFECT = "PULSE_EFFECT"

class Option(BaseModel):
    id: str
    text: str

class Question(BaseModel):
    id: str
    type: QuestionType
    text: str
    options: Optional[List[Option]] = None
    visualFeedback: Optional[VisualFeedback] = None

class QuestInteractionRequest(BaseModel):
    sessionId: str
    questId: str
    answer: str

class QuestResponse(BaseModel):
    sessionId: str
    narrative: str
    question: Optional[Question] = None
    isCompleted: bool = False
    expGained: Optional[int] = 0

class QuestWSEvent(BaseModel):
    event: str
    data: dict

File: frontend/src/types/quest.ts
export interface Question {
  id: string;
  type: 'QUANTITATIVE' | 'SOUL_NARRATIVE';
  text: string;
  options?: { id: string; text: string }[];
  visualFeedback?: string;
}

export interface LevelInfo {
  level: number;
  exp: number;
  next_level_exp: number;
  prev_level_exp: number;
}

export interface FinalResult {
  quest_id: string;
  result_type: string;
  scores: Record<string, number>;
  dimension_scores: Record<string, number>;
  analysis: string;
  level_info: LevelInfo;
  class_id?: string;
  new_talents?: string[];
}

export interface QuestUpdateData {
  narrative?: string;
  questionIndex?: number;
  totalSteps?: number;
  guideMessage?: string;
  question?: Question;
  message?: string;
}

export interface QuestResponse {
  sessionId: string;
  narrative: string;
  question: Question;
  isCompleted: boolean;
}
